tests/app/delivery/test_send_to_providers.py

```diff
@@ -1041,3 +1057,87 @@ def test_is_service_allowed_html(sample_service: Service, notify_api):
+        send_mock.assert_not_called()
+
+        assert Notification.query.get(db_notification.id).status == "virus-scan-failed"
+
+    def test_send_to_providers_fails_if_malware_scan_in_progress(self, sample_email_template, mocker):
+        send_mock = mocker.patch("app.aws_ses_client.send_email", return_value="reference")
+
+        class mock_response:
+            status_code = 428
+
+            def json():
+                return {"av-status": "in_progress"}
+
+        mocker.patch("app.delivery.send_to_providers.document_download_client.check_scan_verdict", return_value=mock_response)
+        personalisation = {"file": document_download_response()}
+
+        db_notification = save_notification(create_notification(template=sample_email_template, personalisation=personalisation))
+
+        with pytest.raises(MalwareScanInProgressException) as e:
+            send_to_providers.send_email_to_provider(db_notification)
+            assert db_notification.id in e.value
+        send_mock.assert_not_called()
+
+        assert Notification.query.get(db_notification.id).status == "pending-virus-check"
+
+    def test_send_to_providers_succeeds_if_malware_verdict_clean(self, sample_email_template, mocker):
+        send_mock = mocker.patch("app.aws_ses_client.send_email", return_value="reference")
+
+        class mock_response:
+            status_code = 200
+
+            def json():
+                return {"av-status": "clean"}
+
+        mocker.patch("app.delivery.send_to_providers.document_download_client.check_scan_verdict", return_value=mock_response)
+        personalisation = {"file": document_download_response()}
+
+        db_notification = save_notification(create_notification(template=sample_email_template, personalisation=personalisation))
+
+        send_to_providers.send_email_to_provider(db_notification)
+        send_mock.assert_called_once()
+
+        assert Notification.query.get(db_notification.id).status == "sending"
+
+    def test_send_to_providers_fails_if_document_download_internal_error(self, sample_email_template, mocker):
+        send_mock = mocker.patch("app.aws_ses_client.send_email", return_value="reference")
+
+        class mock_response:
+            status_code = 404
+
+            def json():
+                return {"av-status": "None"}
+
+        mocker.patch("app.delivery.send_to_providers.document_download_client.check_scan_verdict", return_value=mock_response)
+        personalisation = {"file": document_download_response()}
+
+        db_notification = save_notification(create_notification(template=sample_email_template, personalisation=personalisation))
+
+        with pytest.raises(DocumentDownloadException) as e:
+            send_to_providers.send_email_to_provider(db_notification)
+            assert db_notification.id in e.value
+        send_mock.assert_not_called()
+
+        assert Notification.query.get(db_notification.id).status == "technical-failure"
```

Comment:
Do we have a test for when document download does not fail per se, but the scanning did for some reason? What do we do if malware scanning system is down for some reason? 

I added a [related comment on the ADR](https://github.com/cds-snc/notification-adr/pull/30/files#r1106435018) to the effect of what we'd do if the malware scan is down or returns an actual error.

Reply:
If the scanning doesn't happen, then the `av-status` key will not be written to the object in S3 and dd-api will have a `KeyError` which is handled here: https://github.com/cds-snc/notification-document-download-api/blob/dce41bbcc5090d07552aab14677e56c34f036864/app/utils/store.py#L128-L129

So it will look like the scan is "in progress" and the retry logic will handle the error. So in that case we would retry until we reach the max number of retries. 

If the malware scan verdict comes back as `{"av-status": "error"}` or `"unknown"` or `"unable_to_scan"` then the code treats that the same as a `"clean"` verdict. But the `av-status` gets logged so I was thinking we could make a cloudwatch query or alarm based on that log.